fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/item/v1/FabricItem.java

```diff
@@ -91,4 +91,36 @@ default Multimap<EntityAttribute, EntityAttributeModifier> getAttributeModifiers
 	default boolean isSuitableFor(ItemStack stack, BlockState state) {
 		return ((Item) this).isSuitableFor(state);
 	}
+
+	/**
+	 * Determines if the item will leave a new item stack after being consumed in a recipe.
+	 * (This is also known as "recipe remainder".)
+	 * For example, using a lava bucket in a furnace as fuel will leave an empty bucket.
+	 *
+	 * <p>This is a stack-aware version of {@link Item#hasRecipeRemainder()}.
+	 *
+	 * @param stack the current stack
+	 * @return {@code true} if the item has a recipe remainder
+	 */
+	default boolean hasRecipeRemainder(ItemStack stack) {
```

Comment:
This should be removed and replaced by simply returning `ItemStack.EMPTY` in `getRecipeRemainder`.